### DISCUSSION TITLE
Why .git folder is included in vendor/laravel/framework

### DISCUSSION THREAD

**Discussion by asefsoft** [laravel/framework#43359] [Q&A]
Hello
I just noticed there is a large folder in my vendor and it was a .git folder in `vendor/laravel/framework` contains 73 MB of files in it!

Is it OK and normal?

content of config file is:

```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = https://github.com/laravel/framework.git
	fetch = +refs/heads/*:refs/remotes/origin/*
	pushurl = git@github.com:laravel/framework.git
[branch "9.x"]
	remote = origin
	merge = refs/heads/9.x
[remote "composer"]
	url = https://github.com/laravel/framework.git
	fetch = +refs/heads/*:refs/remotes/composer/*

```

**Comment by henzeb** [comment]
I don't think so. I have checked one of my installations, and I don't have a .git directory there. try deleting your vendor folder and reinstall the dependencies.

**Comment by BrekiTomasson** [comment]
```
bash-5.1# ls -ah
.              ..             CHANGELOG.md   LICENSE.md     README.md      composer.json  src
```

Yeah, I'm not seeing it either. Agree with the previous commenter; delete your vendor folder and then `composer update`.

**Reply by henzeb** [reply]
`composer update` updates the dependencies, `composer install` is better, unless you are planning to update dependencies anyway.

**Comment by Kamandlou** [comment]
I have checked my project and I don't have a .git directory.
Delete your vendor directory and reinstall the dependencies

**Comment by Sophist-UK** [comment]
`composer install` should not download or install the `.git` directory (which contains the entire history of individual changes ever made to Laravel over a period of much more than a decade - hence the size).

You normally get this if you install or update a package from a zip/tar download of the Github repo (rather than from `composer` or a zip/tar download of a Github release).

**Comment by PaolaRuby** [comment]
I have checked my project and I don't have a `.git` directory.

**Comment by vlab22** [comment]
I had this in a turnkey testing server, the `deployer` `deploy:vendors` was installing all the vendors with their .git folders, I don't know why.
I've checked and the `/root/.cache/composer` was huge too.
After running  `composer clear-cache` in the `/root`, the next try `deploy` was ok and the `.git` folders didn't come back.

PS: Everything was correct with `--prefer-dist`, composer.json with config `preferred-install` to `dist` etc.
